### PR TITLE
move build instructions from wiki to source root

### LINKED ISSUE #958: merge repos: move wiki documentation into source folders
**Is your feature request related to a problem? Please describe.**

As a developer I want to read the Linux build instructions **off-line**. Now I need to go to the related GitHub wiki to find those instructions. Also, it is likely that those instructions change over time, which forces me to keep an eye **on that wiki**.

**Describe the solution you'd like**

**Move** the wiki documentation (16 files of about 100kb in total) into a subdirectory **in the source tree**.
# Advantages
1. Build instructions evolve with source code evolutions, which also allows to build old versions of Stellarium without having to find out what version of the instructions correspond to what version of Stellarium. (this problem will not be solved for past Stellarium versions, but for new versions.)  A typical example is the [evolution of the RPi instructions](https://github.com/Stellarium/stellarium/wiki/Raspberry-Pi/_history). **A critical example is the series of changes documented in [this wiki diff](https://github.com/Stellarium/stellarium/wiki/_compare/5766e2c9e48a1f48b28c5c6abde4bd79a9df3c33...9d7e01c61edc2f47bef536f8be0fc8b6d408ef00)**  that **are** related to source code changes 2931ffe0cc928fe4a52b6c87637d66e4b98fa1c3,  ad2e8940255e051e3a51137896beee377dfe797e, b185a58a753503887d70af41eb5be60e3cb3b100 . **These doc wiki changes belong in the source code repository and are critical for correct developer work and communication.**
2. Code changes in the documentation directory inform developers of anything that changed in e.g. the compilation instructions. (Such changes could e.g. be related to updates in e.g Qt libraries, to one-time changes to be applied to the compilation environment, etc)
3. I will also automatically (via diffs) be informed of new as well as solved "[Common Problems for the current version](https://github.com/Stellarium/stellarium/wiki/Common-Problems-for-the-current-version)". Observe that the issues reported in that page are closely related to code base evolution - in other words, this is related to versioning.
4. There is only one repository to track.

# Disadvantages
1. GitHub users cannot contribute to documentation pages without submitting a pull request via their own fork (a process that is supported by GitHub). **This is not a real issue** because the history of the wiki shows that the vast majority of changes were done by developers, and some changes were part of efforts that also involved changes in source code. Also, this will prevent random/spam changes such as [this](https://github.com/Stellarium/stellarium/wiki/_compare/5426edaaa845a2e6819c031b135eb2487556cf67...4e6eeea45e85d8b5e2e60745e2225e25549c16d6) .
2. There is no fancy table of content side panel

**Describe alternatives you've considered**
There is no alternative but to regularly consult the documentation wiki.

**Additional context**
As an example, https://github.com/Stellarium/stellarium/issues/211#issuecomment-581146615 should no longer occur because a developer - assumming he/she is aware of this entry - could solve an issue (as was done in ?) and at the same time remove such an entry.

### PR DESCRIPTION
**Pull request by axd1967**
Ensure code build instructions closely follow source root changes.

### Description

Replaces wiki pages

* https://github.com/Stellarium/stellarium/wiki/Code-Testing
* https://github.com/Stellarium/stellarium/wiki/Compilation-on-Linux
* https://github.com/Stellarium/stellarium/wiki/Compilation-on-macOS
* https://github.com/Stellarium/stellarium/wiki/Compilation-on-Windows
* https://github.com/Stellarium/stellarium/wiki/Linux-build-dependencies
* https://github.com/Stellarium/stellarium/wiki/Raspberry-Pi

Fixes #958

### CODE REVIEW

**Comment by gzotti** [comment]
There are some hyperlinks esp. in the RaspberryPi part. These should no longer point to external .md files. Either change to in-doc hyperlinks, or (if that is not available) into phrases like "see ... above".

File: BUILDING.md [hunk 1]
```diff
@@ -0,0 +1,623 @@
...
+        --with-egl-platforms=x11,drm
+    make
+    sudo make install
+
+## Notes for building Stellarium
+
+You may want to install a swap file, otherwise you may run out of virtual memory when compiling
+[https://www.cyberciti.biz/faq/linux-add-a-swap-file-howto/](https://www.cyberciti.biz/faq/linux-add-a-swap-file-howto/)
+
+Then, make sure to install the [build requirements](#lbuild-dependencies) and follow the [building instructions for Linux](#linux).
```
Comment (gzotti) [inline BUILDING.md]: I randomly clicked on a few hyperlinks in an MD editor. This failed, program crashed.

Comment (axd1967) [reply BUILDING.md]: ok. what editor was that, please?

Comment (axd1967) [reply BUILDING.md]: it's possibly the GMD syntax that gives this issue.

Comment (gzotti) [reply BUILDING.md]: "Markdown Editor" by Poppen's Lab, my first usable contact with MD. I accept recommendations for better tools as long as they are free.

**Comment by axd1967** [comment]
To be honest, I don't use a Mardown editor (and a dev will just browse the text or browse the GitHub page).  Often there are good plugins for IDE's: for example in PyCharm the links works fine.

image

I couldn't find that editor yet.

(I haven't set up a good Stellarium dev environment yet to evaluate markdown plugins.)

**Comment by axd1967** [comment]
@gzotti 
Can we proceed with the merge? We cannot test every possible code editor. The essence is that the text can be read in a simple text editor. Essence is that a developer should have a single place where all information is present; avoid two places that have to be maintained in parallel, which is confusing.

**Comment by gzotti** [comment]
In the last weeks other things were more important than where to place build instructions. Of course there should only be one place. If we combine all docs to one, who shall go through the text and reduce e.g. git duplicates or harmonize instructions? E.g. Raspberry notes may be a part of Linux. 
@alex-w any comments?

**Comment by alex-w** [comment]
Just copying few documents into one big document is not very good solution.

For example you are wrote in the first paragraph:
> Each time Stellarium is released, the source code is released in Github's packaging system.  Building source code that is released in this way should give you a working copy of Stellarium which is functionally identical to the binaries for that release. 

and in paragraph 15 (Qt versions) user may see:
> Stellarium 0.13.0 required Qt version 5.1.0 or above for building....

Seriously? You proposed moving and restructure the building instruction for offline usage and those instructions should be actual. Sorry, but this version of instructions is not actual.

If you began using Qt 5.5 for Linux as example, then you need use same version for macOS and Windows too. At the moment minimal version Qt for building master version of Stellarium is 5.6.0, but we recommended using Qt 5.9 or, better, Qt 5.12 (both - LTS) for building planetarium.

Other problem: the BUILDING.md file missing important part for building the binary package from source code - configuration options - http://stellarium.sourceforge.net/wiki/index.php/Configuring_Build_Options

Yes, the list of options is not actual now and it should be revised. Of course part of instructions, who related to Raspberry Pi need move to Linux block with remarks.

Many parts of text are redundant by historical reasons and it should be revised too - so, final document need editing and proofreading!

**Comment by axd1967** [comment]
PLEASE! I cannot 

* move the instructions
* AND update them!

I have no vision on everything that changed, and cannot know it completely.

Can we PLEASE split this into TWO steps:
1. move the instructions
2. update the instructions


by blocking this PR because the instructions are outdated is a bad idea, this will never close this issue.

I never proposed to **UPDATE** the instructions.

**Comment by alex-w** [comment]
@axd1967, sorry, but who are wrote advantages list in the issue #958? O_O

**Comment by axd1967** [comment]
All I write is 
> Move the wiki documentation (16 files of about 100kb in total) into a subdirectory in the source tree.

**Comment by gzotti** [comment]
I think this is also superseded and can be closed.

**Pull request closed without merging**